frontend/packages/network-attachment-definition-plugin/src/plugin.ts

```diff
@@ -0,0 +1,51 @@
+import * as _ from 'lodash';
+import * as models from '@console/shared/src/models';
+import {
+  Plugin,
+  ResourceNSNavItem,
+  ResourceListPage,
+  ModelFeatureFlag,
+  YAMLTemplate,
+  ModelDefinition,
+  RoutePage,
+} from '@console/plugin-sdk';
+
+type ConsumedExtensions =
+  | ResourceNSNavItem
+  | ResourceListPage
+  | ModelFeatureFlag
+  | YAMLTemplate
+  | ModelDefinition
+  | RoutePage;
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'ModelDefinition',
+    properties: {
+      models: _.values(models),
```

Comment:
seeing as this plugin contributes models from `console-shared` is weird. I'd rather put the NAD model to `network-attachment-definition` plugin and `kubevirt` plugin will depend on it.

Reply:
Done!

Reply:
Exactly as @rawagner said, each model definition should be "owned" by a specific plugin, which adds them to Console via `ModelDefinition` extension.

Putting model definitions into `console-shared` doesn't make sense.

@pcbailey Please do the following:
- `git mv` from `console-shared/src/models/index.ts` to `network-attachment-definition-plugin/src/models/index.ts`
- in `network-attachment-definition-plugin/src/plugin.ts`: `import * as models from './models';`
- add `network-attachment-definition-plugin/src/index.ts` with `export * from './models';`
- in `packages/network-attachment-definition-plugin/package.json` add `"main": "src/index.ts",`

With these changes, other packages can do

```ts
import { NetworkAttachmentDefinitionModel } from '@console/network-attachment-definition-plugin';
```

Any package that wishes to expose its code to other packages _should_ define a `main` field in its `package.json` file, pointing to `src/index.ts` which exports stuff needed by other packages.